internal/xds/translator/route.go

```diff
@@ -82,18 +82,15 @@ func buildXdsRoute(httpRoute *ir.HTTPRoute, listener *listener.Listener) *routev
 func buildXdsRouteMatch(pathMatch *ir.StringMatch, headerMatches []*ir.StringMatch, queryParamMatches []*ir.StringMatch) *routev3.RouteMatch {
 	outMatch := &routev3.RouteMatch{}
 
-	// Return early with a prefix match to '/' if no matches are specified
-	if pathMatch == nil && len(headerMatches) == 0 && len(queryParamMatches) == 0 {
+	// Add a prefix match to '/' if no matches are specified
+	if pathMatch == nil {
```

Comment:
What happens if a `""` prefix is supplied?

Reply:
case looks unlikely since by default it is set to `/` 
https://github.com/kubernetes-sigs/gateway-api/blob/4ff6208a6a4238792e7477a03400933f5ae26c42/apis/v1beta1/httproute_types.go#L308